lib/features/modeling/behavior/RootElementReferenceBehavior.js

```diff
@@ -0,0 +1,165 @@
+  }
+
+  // create shape
+  this.executed('shape.create', function(context) {
+    var shape = context.shape;
+
+    if (!hasAnyEventDefinition(shape, [
+      'bpmn:ErrorEventDefinition',
+      'bpmn:EscalationEventDefinition',
+      'bpmn:MessageEventDefinition',
+      'bpmn:SignalEventDefinition'
+    ])) {
+      return;
+    }
+
+    var businessObject = getBusinessObject(shape),
+        eventDefinitions = businessObject.get('eventDefinitions'),
+        eventDefinition = eventDefinitions[ 0 ],
+        rootElement = getRootElementReferenced(eventDefinition),
+        rootElements;
+
+    if (rootElement && !hasRootElement(rootElement)) {
+      rootElements = bpmnjs.getDefinitions().get('rootElements');
+
+      // add root element
+      collectionAdd(rootElements, rootElement);
+
+      context.addedRootElement = rootElement;
+    }
+  }, true);
+
+  this.reverted('shape.create', function(context) {
+    var addedRootElement = context.addedRootElement;
+
+    if (!addedRootElement) {
+      return;
+    }
+
+    var rootElements = bpmnjs.getDefinitions().get('rootElements');
+
+    // remove root element
+    collectionRemove(rootElements, addedRootElement);
+  }, true);
+
+  eventBus.on('copyPaste.copyElement', function(context) {
+    var descriptor = context.descriptor,
+        element = context.element;
+
+    if (!hasAnyEventDefinition(element, [
+      'bpmn:ErrorEventDefinition',
+      'bpmn:EscalationEventDefinition',
+      'bpmn:MessageEventDefinition',
+      'bpmn:SignalEventDefinition'
+    ])) {
+      return;
+    }
+
+    var businessObject = getBusinessObject(element),
+        eventDefinitions = businessObject.get('eventDefinitions'),
+        eventDefinition = eventDefinitions[ 0 ],
+        rootElement = getRootElementReferenced(eventDefinition);
+
+    if (rootElement) {
+      descriptor.referencedRootElement = rootElement;
+    }
+  });
+
+  eventBus.on('copyPaste.pasteElement', LOW_PRIORITY, function(context) {
+    var descriptor = context.descriptor,
+        businessObject = descriptor.businessObject;
+
+    if (!hasAnyEventDefinition(businessObject, [
+      'bpmn:ErrorEventDefinition',
+      'bpmn:EscalationEventDefinition',
+      'bpmn:MessageEventDefinition',
+      'bpmn:SignalEventDefinition'
+    ])) {
+      return;
+    }
+
+    var eventDefinitions = businessObject.get('eventDefinitions'),
+        eventDefinition = eventDefinitions[ 0 ],
+        referencedRootElement = descriptor.referencedRootElement;
+
+    if (!referencedRootElement) {
+      return;
+    }
+
+    eventDefinition.set(getRootElementReferencePropertyName(eventDefinition), referencedRootElement);
+  });
+}
+
+RootElementReferenceBehavior.$inject = [
+  'bpmnjs',
+  'eventBus',
+  'injector'
+];
+
+inherits(RootElementReferenceBehavior, CommandInterceptor);
+
+// helpers //////////
+
+function hasAnyEventDefinition(element, types) {
+  if (!isArray) {

```

Comment:
This looks broken and probably is.

Reply:
How did this happen? 🤣

Reply:
You wonder how I've spotted this? Looked into code coverage :o).